src/main/java/org/craftercms/engine/navigation/impl/DefaultItemConverter.java

```diff
@@ -99,6 +108,23 @@ protected String getNavigationUrl(SiteItem siteItem) {
         return urlTransformationService.transform(storeUrlToRenderUrlTransformerName, siteItem.getStoreUrl());
     }
 
+    protected Map<String, String> getAdditionalAttributes(SiteItem siteItem) {
+        SiteContext siteContext = SiteContext.getCurrent();
+        Configuration siteConfig = siteContext.getConfig();
+        if(siteConfig != null && siteConfig.containsKey(KEY_NAV_ADDITIONAL_FIELDS)) {
```

Comment:
Please use the org.craftercms.engine.properties.SiteProperties, since it contains Engine's common site config properties.